src/wallet/scriptpubkeyman.h

```diff
@@ -110,36 +110,37 @@ class CKeyPool
     CKeyPool();
     CKeyPool(const CPubKey& vchPubKeyIn, bool internalIn);
 
-    ADD_SERIALIZE_METHODS;
+    template<typename Stream>
+    void Serialize(Stream& s) const
+    {
+        int nVersion = s.GetVersion();
+        if (!(s.GetType() & SER_GETHASH)) {
+            s << nVersion;
+        }
+        s << nTime << vchPubKey << fInternal << m_pre_split;
+    }
 
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
         int nVersion = s.GetVersion();
```

Comment:
ef17c03 is this new change from 309a653 intentional (needed for deserializing)?
```diff
-        int nVersion;
+        int nVersion = s.GetVersion();
```

Reply:
I doubt it's needed, as I don't think this class is ever deserialized with SER_GETHASH. However, if it were ever invoked that way, the old code was UB (nVersion would be used without being initialized).

Reply:
Thank you. Good catch and reminder.